crates/sui-core/src/validator_client_monitor/monitor.rs

```diff
@@ -165,8 +165,7 @@ impl<A: Clone> ValidatorClientMonitor<A> {
     ///
     /// This method is called periodically after health checks complete to update
     /// the cached validator scores.
-    fn update_cached_scores(&self) {
-        let authority_agg = self.authority_aggregator.load();
+    fn update_cached_scores(&self, authority_agg: &AuthorityAggregator<A>) {
```

Comment:
Passing the `authority_agg` isn't necessary for the fix, but to allows test the scenario of missing stats for a validator set (and confirm that no error/panic will occur)